mantidimaging/gui/windows/spectrum_viewer/test/presenter_test.py

```diff
@@ -299,16 +300,60 @@ def test_WHEN_tof_unit_selected_THEN_model_mode_changes(self, mode_text, expecte
         self.presenter.handle_tof_unit_change_via_menu()
         self.assertEqual(self.presenter.model.tof_mode, expected_mode)
 
+    @parameterized.expand([
+        (None, ToFUnitMode.IMAGE_NUMBER),
+        (np.arange(1, 10), ToFUnitMode.WAVELENGTH),
+    ])
     @mock.patch("mantidimaging.gui.windows.spectrum_viewer.model.SpectrumViewerWindowModel.get_stack_time_of_flight")
-    def test_WHEN_no_spectrum_data_THEN_mode_is_image_index(self, get_stack_time_of_flight):
+    def test_WHEN_data_loaded_THEN_relevant_mode_set(self, tof_data, expected_tof_mode, get_stack_time_of_flight):
         self.presenter.model.set_stack(generate_images())
         self.presenter.get_dataset_id_for_stack = mock.Mock(return_value=uuid.uuid4())
         self.presenter.main_window.get_stack = mock.Mock(return_value=generate_images())
-        get_stack_time_of_flight.return_value = None
+        get_stack_time_of_flight.return_value = tof_data
         self.view.tof_units_mode = "Wavelength"
         self.presenter.refresh_spectrum_plot = mock.Mock()
         self.presenter.handle_sample_change(uuid.uuid4())
-        self.assertEqual(self.presenter.model.tof_mode, ToFUnitMode.IMAGE_NUMBER)
+        self.assertEqual(self.presenter.model.tof_mode, expected_tof_mode)
+
+    @parameterized.expand([
+        (None, "Image Index", ToFUnitMode.IMAGE_NUMBER, np.arange(1, 10), [mock.call(False),
```

Comment:
A small tweak that reduces the wrapping. If you change `[mock.call(False), mock.call(True)],` to `[False, True]`, then you can turn the into `call`s in the method
`expected_calls = [mock.call(b) for b in expected_calls]`